File: 2022-2023/sourcecode/lab/lab3/QuickYogaAppJava/app/src/main/java/pl/edu/uwr/pum/quickyogaappjava/YogaPoses.java
package pl.edu.uwr.pum.quickyogaappjava;

import java.util.ArrayList;

public final class YogaPoses {
    private YogaPoses(){}

    public static ArrayList<YogaPose> getYogaPoses(){
        int[] images = {
                R.drawable.e1w,
                R.drawable.e2w,
                R.drawable.e3w,
                R.drawable.e4w,
                R.drawable.e5w,
                R.drawable.e6w,
                R.drawable.e7w,
                R.drawable.e8w,
                R.drawable.e9w,
                R.drawable.e10w,
                R.drawable.e11w,
                R.drawable.e12w,
        };

        String[] names = {"Cobra", "Lord of the Dance", "Low Lunge", "Dancer",
                "Wide Leg Forward Bend", "Warrior", "Lord Of The Dance Full",
                "Squatting Toe Balance", "Lord Of The Dance Revolved",
                "Wheel, One Legged", "Shoulder Stand, Split", "Warrior, Reverse"};

        ArrayList<YogaPose> yogaPoses = new ArrayList<>(12);
        for(int i = 0; i < 12; i++){
            yogaPoses.add(
                    new YogaPose(
                            i,
                            names[i],
                            images[i]
                    )
            );
        }
        return yogaPoses;
    }
}

File: 2022-2023/sourcecode/lab/lab3/QuickYogaAppKotlin/app/src/main/java/pl/edu/uwr/pum/quickyogaappkotlin/YogaPoses.kt
package pl.edu.uwr.pum.quickyogaappkotlin

object YogaPoses {
    val yogaPoses: ArrayList<YogaPose>
        get() {
            val images = intArrayOf(
                R.drawable.e1w,
                R.drawable.e2w,
                R.drawable.e3w,
                R.drawable.e4w,
                R.drawable.e5w,
                R.drawable.e6w,
                R.drawable.e7w,
                R.drawable.e8w,
                R.drawable.e9w,
                R.drawable.e10w,
                R.drawable.e11w,
                R.drawable.e12w
            )
            val names = arrayOf(
                "Cobra", "Lord of the Dance", "Low Lunge", "Dancer",
                "Wide Leg Forward Bend", "Warrior", "Lord Of The Dance Full",
                "Squatting Toe Balance", "Lord Of The Dance Revolved",
                "Wheel, One Legged", "Shoulder Stand, Split", "Warrior, Reverse"
            )
            val yogaPoses: ArrayList<YogaPose> = ArrayList(12)
            for (i in 0..11) {
                yogaPoses.add(
                    YogaPose(
                        i,
                        names[i],
                        images[i]
                    )
                )
            }
            return yogaPoses
        }
}